tidy3d/components/data/dataset.py

```diff
@@ -550,14 +550,41 @@ def _double_type(self):
         """Corresponding double data type."""
         return np.complex128 if self.is_complex else np.float64
 
-    @pd.validator("cells", always=True)
-    def match_cells_to_vtk_type(cls, val):
-        """Check that cell connections does not have duplicate points."""
-        if vtk is None:
-            return val
+    @pd.validator("points", always=True)
+    def points_right_dims(cls, val):
+        """Check that point coordinates have the right dimensionality."""
+        # currently support only the standard axis ordering, that is 01(2)
+        axis_coords_expected = np.arange(cls._point_dims())
+        axis_coords_given = val.axis.data
+        if np.any(axis_coords_given != axis_coords_expected):
+            raise ValidationError(
+                f"Points array is expected to have {axis_coords_expected} coord values along 'axis'"
+                f" (given: {axis_coords_given})."
+            )
+        return val
 
-        # using val.astype(np.int32/64) directly causes issues when dataarray are later checked ==
-        return CellDataArray(val.data.astype(vtk["id_type"], copy=False), coords=val.coords)
+    @pd.validator("points", always=True)
+    def points_right_indexing(cls, val):
+        """Check that points are indexed corrrectly."""
+        indices_expected = np.arange(len(val.data))
+        indices_given = val.index.data
+        if np.any(indices_expected != indices_given):
+            raise ValidationError(
+                "Points array is expected to be indexed in a simple fashion (0, 1, 2, ...)."
```

Comment:
Suggest adding a little bit more explanation in case it's not clear how to fix this validator.
```
"'points.index' expects a simple array of indices for each data, eg. '(0, 1, 2, ...)' or 'np.arange(len(points.data))'. "
```

Same comments in the validator below

Reply:
Also, I'm slightly confused why we need this `index` as a field at all if it seems it only accepts one possible value?

Reply:
`index` here is one of the xarray coordinates of `PointDataArray`. But, yeah, we don't need this capability of xarray's to provide flexible coordinate values, it was primarily used to be able to easily write this data in hdf5 using our existing tools. This check is more to prevent users from providing more complicated `index` values and expecting that they would be taken into account in our unstructured datasets. If there is a way to force `PointDataArray` always have certain coordinates, that would be easier. When defining, the easiest way to provide what's needed is to use `PointDataArray(data, dims=["index", "axis"])`

Reply:
we could consider extending our unstructured datasets to take into account any `index` values, but that probably will not add any useful functionality